app/commands/v2/import.rb

```diff
@@ -0,0 +1,232 @@
+      def call
+        unless UuidValidator.valid?(payload[:content_id])
+          raise CommandError.new(
+            code: 422,
+            error_details: {
+              error: {
+                code: 422,
+                message: "Content id not valid",
+                fields: "content_id",
+              }
+            }
+          )
+        end
+
+        previous_content_store_base_paths = get_base_path_content_store_pairs
+
+        delete_all(payload[:content_id])
+        content_items = payload[:history].map.with_index do |event, index|
+          create_content_item(event, index, payload[:content_id])
+        end
+
+        not_draft_content_item = nil
+        content_items.reverse_each do |content_item|
+          if not_draft_content_item.present?
+            content_item.supersede
+          elsif content_item.state != "draft"
+            not_draft_content_item = content_item
+          end
+        end
+
+        new_content_store_base_paths = get_base_path_content_store_pairs
+
+        after_transaction_commit do
+          Commands::V2::RepresentDownstream.new.call(
+            payload[:content_id],
+            true
+          )
+
+          delete_content_from_live_content_store(
+            previous_content_store_base_paths,
+            new_content_store_base_paths
+          )
+
+          delete_content_from_draft_content_store(
+            previous_content_store_base_paths,
+            new_content_store_base_paths
+          )
+        end
+
+        Success.new(content_id: payload[:content_id])
+      end
+
+    private
+
+      def create_content_item(history_entry, index, content_id)
+        validate_history_entry(history_entry, index)
+
+        create_content_item_service = Services::CreateContentItem.new(
+          payload: history_entry.except("states").merge(
+            content_id: content_id,
+            user_facing_version: index + 1,
+            state: "draft",
+          ),
+          lock_version: index + 1,
+        )
+
+        create_content_item_service.create_content_item do |content_item|
+          update_content_item_state_information(content_item, history_entry[:states])
+        end
+      end
+
+      def update_content_item_state_information(content_item, states)
+        states.each do |state|
+          case state[:name]
+          when "unpublished"
+            content_item.unpublish(
+              state.slice(
+                *%i(type explanation alternative_path unpublished_at)
+              )
+            )
+          when "superseded"
+            content_item.supersede
+          when "published"
+            content_item.publish
+          when "draft"
+            content_item.state = "draft"
+          else
+            raise CommandError.new(
+              code: 422,
+              message: "Unrecognised state: #{state[:name]}."
+            )
+          end
+        end
+      end
+
+      def validate_history_entry(history_entry, index)
```

Comment:
I've added a test to check that no invalid states slip through in db2365b . I'm tempted to not add any explicit validation to the import endpoint, as it doesn't feel like the right place.